### ISSUE TITLE
issue with bluetooth 5.3 and remote bypass

### ISSUE THREAD

**Issue by jefwoe** [pivo-inc/pivo-pro-sdk-testapp-ios#6]
We have been having reports of our users having issues with our app on iPhone 15, so we did some investigation. It appears as though the issue is related to the new version of bluetooth first introduced on the iPhone 14 (Bluetooth 5.3). I did some testing and sure enough, if I enable remote bypass on the iPhone 15, the response of the remote is extremely slow (more than a few seconds). If I disable remote bypass it works quickly, as you would expect.

**Comment by 3i-kevin** [comment]
Hi @jefwoe 
Could you specify the iOS version which the issue happen? 
And also for our better understanding, can you elaborate the use case of your app when you enable/disable remote bypass? What is the role of the remote controller?

**Comment by jefwoe** [comment]
Its iOS 17. The remote is still used as a remote, but we change the behavior of various buttons depending on the mode of our app.

On Oct 18, 2023, at 11:53 PM, Kevin N. Santoso ***@***.***> wrote:



Hi @jefwoe<https://github.com/jefwoe>
Could you specify the iOS version which the issue happen?
And also for our better understanding, can you elaborate the use case of your app when you enable/disable remote bypass? What is the role of the remote controller?

**Comment by 3i-kevin** [comment]
Could you tell us what is the Pod device type that you and your users use?

Let me clarify the behavior:
1. Remote Bypass OFF: when pressing a button, for example "turn right", the pod rotates immediately and the app receives notification quickly.
2. Remote Bypass ON: when pressing a button, the pod does not rotate (as expected) however the app receives notification very late.

Am I correct?

**Comment by jefwoe** [comment]
You are correct about the behavior. I have tested it and confirmed the issue on the Pivo Max. I believe Derek tested it on an older device.

On Oct 26, 2023, at 6:46 PM, Kevin N. Santoso ***@***.***> wrote:



Could you tell us what is the Pod device type that you and your users use?

Let me clarify the behavior:

  1.  Remote Bypass OFF: when pressing a button, for example "turn right", the pod rotates immediately and the app receives notification quickly.
  2.  Remote Bypass ON: when pressing a button, the pod does not rotate (as expected) however the app receives notification very late.

Am I correct?

**Comment by 3i-kevin** [comment]
@jefwoe  I apologize for the long wait. We have tried to simulate your use case but we could not reproduce the issue that you reported. When we test using the TestApp, the remote controller callback arrives immediately. Our testing condition covers:
- Bypass ON and OFF
- Pivo pod black & silver, Pivo Max
- iPhone 15 Pro Max (iOS 17.1.1)

Would it be possible for you to try the provided TestApp and check if you experience the same issue as in your app? Also, could you tell us your development environment, such as 
- Macbook details
- Xcode version
- Target iOS build
Thank you
